tests/RobotFramework/config_management/child_conf_mgmt_plugin.robot

```diff
@@ -0,0 +1,333 @@
+    ${child_id}    Get From List    ${id}    1
+    Set Suite Variable    ${child_id}
+Check parent child relationship
+    ${auth}=    Create List    ${user}    ${pass}
+    Create Session    API_Testing    https://${url_tedge}    auth=${auth}
+    ${Get_Response}=    GET On Session    API_Testing    /inventory/managedObjects/${parent_id}/childDevices/${child_id}    expected_status=200    #Command: c8y inventory children get --childType device --id ${parent_id} --child ${child_id}
+Reconnect c8y
+    ${rc}=    Execute Command    sudo tedge disconnect c8y    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0}   
+    ${rc}=    Execute Command    sudo tedge connect c8y    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0} 
+Restart Configuration plugin
+    ${rc}=    Execute Command    sudo systemctl stop c8y-configuration-plugin.service    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0} 
+    ${rc}=    Execute Command    sudo systemctl start c8y-configuration-plugin.service    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0} 
+Child device delete configuration files
+    Child Connection
+    Execute Command    sudo rm config1
+    Execute Command    sudo rm c8y-configuration-plugin
+    Close Connection
+GET Parent name
+    ${auth}=    Create List    ${user}    ${pass}
+    Create Session    API_Testing    https://${url_tedge}    auth=${auth}
+    ${Get_Response}=    GET On Session    API_Testing    /identity/globalIds/${parent_id}/externalIds
+    ${json_response}=    Set Variable    ${Get_Response.json()}
+    @{pd_name}=    Get Value From Json    ${json_response}    $..externalId
+    ${pardev_name}    Get From List    ${pd_name}    0
+    Set Suite Variable    ${pardev_name}
+GET Child name
+    ${auth}=    Create List    ${user}    ${pass}
+    Create Session    API_Testing    https://${url_tedge}    auth=${auth}
+    ${Get_Response}=    GET On Session    API_Testing    /inventory/managedObjects?fragmentType\=c8y_IsDevice
+    ${json_response}=    Set Variable    ${Get_Response.json()}  
+    @{name}=    Get Value From Json    ${json_response}    $..managedObject.name
+    ${child_name}    Get From List    ${name}    0
+    Set Suite Variable    ${child_name}
+Validate child Name
+    Should Be Equal    ${CHILD}     ${child_name}
+Startup child device
+    Child Connection
+    ${rc}=    Execute Command    printf ${config} > c8y-configuration-plugin    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0}
+    Write    curl -X PUT http://${PARENT_IP}:${HTTP_PORT}/tedge/file-transfer/${CHILD}/c8y-configuration-plugin \\            #Folder will be created /var/tedge/file-transfer
+    Write   --data-binary @- << EOF
+    Write   files = [
+    Write        { path = '/home/pi/config1', type = 'config1' },
+    Write    ]
+    Write  EOF 
+    Execute Command    sudo apt-get install mosquitto-clients -y
```

Comment:
This installation might be better in `Prerequisite Child Device` step, isn't it?

Reply:
In that case the Documentation has to be changed as well, in the documentation this is part of the bootstrapping/startup procedure. I try to keep the tests as close to docs as possible.

Reply:
Well... The bootstrapping or even the handling of those requests doesn't necessarily need `mosquitto-clients`. In a realistic use-case, there will be some daemon on the child device that handles these MQTT interactions and that daemon program is free to choose any MQTT library for that purpose. The examples in the documentation just uses `mosquitto-clients` itself just as an example.

The point that I was trying make is that the MQTT library(`mosquitto-clients` in these tests) is not a requirement for the bootstrapping alone, but required for further config operation handlings as well. So, it's more like a prerequisite for the entire `child device agent` software itself.

Here it's not making much difference whether the installation happens in `Prerequisites` phase or here as this bootstrap is the first interaction that uses this library. But, I hope you got my point.

Reply:
We just have discussion with @reubenmiller, will add Suite setup and teardown, will include all "preparations" there